resources/js/components/actions/ConfirmableAction.vue

```diff
@@ -0,0 +1,130 @@
+<script setup>
+import { ref, computed } from 'vue';
+import PublishFields from '../publish/Fields.vue';
+import { requireElevatedSessionIf } from '@statamic/components/elevated-sessions';
+
+const props = defineProps({
+    action: { type: Object, required: true },
+    selections: { type: Number, required: true },
+    errors: { type: Object },
+    isDirty: { type: Boolean, default: false },
+});
+
+const emit = defineEmits(['confirmed']);
+
+let confirming = ref(false);
+let running = ref(false);
+let fieldset = ref({ tabs: [{ fields: props.action.fields }] });
+let values = ref(props.action.values);
+
+let confirmationText = computed(() => {
+    if (!props.action.confirmationText) return;
+    return __n(props.action.confirmationText, props.selections);
+});
+
+let warningText = computed(() => {
+    if (!props.action.warningText) return;
+    return __n(props.action.warningText, props.selections);
+});
+
+let dirtyText = computed(() => {
+    if (!props.isDirty) return;
+    return __(props.action.dirtyWarningText);
+});
+
+let showDirtyWarning = computed(() => {
+    return props.isDirty && props.action.dirtyWarningText && !props.action.bypassesDirtyWarning;
+});
+
+let runButtonText = computed(() => {
+    return __n(props.action.buttonText, props.selections);
+});
+
+// TODO: How do we convert these to composition API?
+//
+// created() {
+//     this.$events.$on('reset-action-modals', this.reset);
+// },
+//
+// unmounted() {
+//     this.$events.$off('reset-action-modals', this.reset);
+// },
```

Comment:
Instead of `this.$events`, change to `Statamic.$events`. Merge the `ui` branch back into this one. I only just recently made it possible.